### PR TITLE
chore(auth): general clean up for credentialsProvider and also rename Credentials from sdk

### PR DESCRIPTION
**Pull request by Samaritan1011001**
#### Description of changes

- General clean up for the CognitoCredentialsProvider
- IdentityIdStore's loadIdentityId now returns Identity or null 
- Rename Credentials to AWSCredentialsIdentity as a BB item

#### Description of how you validated changes
Works as expected for the following scenarios on a React app,
- guest user fetchAuthSession
- guest user -> sign In -> fetchAuthSession -> sign out -> fetchAuthSession
- sign in -> fetchAuthSession -> sign out

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/core/src/clients/types/aws.ts [hunk 1]
```diff
@@ -5,7 +5,10 @@ import { MetadataBearer } from '@aws-sdk/types';
 import { Endpoint } from './core';
 import { HttpResponse } from './http';
 
-export type { Credentials, MetadataBearer } from '@aws-sdk/types';
+export type {
+	AwsCredentialIdentity as Credentials,
+	MetadataBearer,
+} from '@aws-sdk/types';
```
Comment (elorzafe) [inline packages/core/src/clients/types/aws.ts:11]: are we doing the thing that Allan did on v5 to generate those and have `@aws-sdk/types` as dev dependency

**Review by elorzafe — APPROVED** [review]
Looks good!
qq: what about generating the types instead of making it a dependency

File: packages/core/src/clients/types/aws.ts (hunk 1 above)
Comment (israx) [reply packages/core/src/clients/types/aws.ts:11]: +1

**Review by israx — APPROVED** [review]
I think we should add a todo to auto generate the types from the SDK. Other than that, LGTM.

File: packages/core/src/clients/types/aws.ts (hunk 1 above)
Comment (Samaritan1011001) [reply packages/core/src/clients/types/aws.ts:11]: Okay, we can create a follow up PR for this. Thanks

**Pull request merged**